R/SpectrumList.R

```diff
@@ -0,0 +1,128 @@
+#' @name SpectrumList
+#'
+#' @aliases SpectrumList-class show,SpectrumList-method
+#'
+#' @title List of Spectrum objects along with annotations
+#'
+#' @description
+#'
+#' `SpectrumList` objects allow to collect one or more [Spectrum] object(s)
+#' ([Spectrum1] or [Spectrum2]) in a `list`-like structure with the additional
+#' possibility to add arbitrary annotations to each individual [Spectrum]
+#' object. These can be accessed/set with the [mcols] method.
+#'
+#' @details
+#'
+#' `SpectrumList` inherits all methods from the [SimpleList] class of the
+#' `S4Vectors` package. This includes `lapply` and other data manipulation
+#' and subsetting operations.
+#'
+#' @param object a `SpectrumList` object.
+#'
+#' @param x a `SpectrumList` object.
+#' 
+#' @md
+#'
+#' @rdname SpectrumList
+NULL
+
+.SpectrumList <- setClass("SpectrumList",
```

Comment:
Of course we will add it to `DataClasses.R`. I simply copied the files over from `CompoundDb.R`.

Reply:
In the end it depends if we agree to add the object or not. If so I would add it to `DataClasses.R` (I would also call it `Spectra` instead of `SpectrumList`) and would address also all points raised by Sebastian.

Reply:
Fine then. Go ahead.